gradio/queueing.py

```diff
@@ -461,74 +459,6 @@ def get_status(self) -> EstimationMessage:
             queue_size=len(self),
         )
 
-    async def call_prediction(self, events: list[Event], batch: bool) -> dict:
-        body = events[0].data
-        if body is None:
-            raise ValueError("No event data")
-        username = events[0].username
-        body.event_id = events[0]._id if not batch else None
-        try:
-            body.request = events[0].request
-        except ValueError:
-            pass
-
-        if batch:
-            body.data = list(zip(*[event.data.data for event in events if event.data]))
-            body.request = events[0].request
-            body.batched = True
-
-        app = self.server_app
-        if app is None:
-            raise Exception("Server app has not been set.")
-        api_name = "predict"
-
-        fn_index_inferred = route_utils.infer_fn_index(
-            app=app, api_name=api_name, body=body
-        )
-
-        gr_request = route_utils.compile_gr_request(
-            app=app,
-            body=body,
-            fn_index_inferred=fn_index_inferred,
-            username=username,
-            request=None,
-        )
-        assert body.request is not None  # noqa: S101
-        root_path = route_utils.get_root_url(
-            request=body.request, route_path="/queue/join", root_path=app.root_path
-        )
-        try:
-            output = await route_utils.call_process_api(
-                app=app,
-                body=body,
-                gr_request=gr_request,
-                fn_index_inferred=fn_index_inferred,
-                root_path=root_path,
-            )
-        except Exception as error:
-            show_error = app.get_blocks().show_error or isinstance(error, Error)
-            traceback.print_exc()
-            raise Exception(str(error) if show_error else None) from error
-
-        # To emulate the HTTP response from the predict API,
```

Comment:
I think this was added for lite so just want to make sure it still works

Reply:
@whitphx can you explain what this postprocessing does and if its still needed for lite? It is very expensive to run.

Reply:
This is not only for Lite.
This was added in https://github.com/gradio-app/gradio/pull/5165/files#diff-cab37b7b6ce45220156832fd4ea1597bedcb26f5cfc43ab5c678fc8a586a0232L379 where HTTP requests are replaced with the direct function calls of the prediction endpoint handler.
As commented in the code, [FastAPI automatically converts what returned from the endpoint handler into a JSON response](https://fastapi.tiangolo.com/advanced/response-directly/) ("By default, FastAPI would automatically convert that return value to JSON") in case of HTTP access, but with this change, this FastAPI's JSON conversion would be skipped, so I added this part to make the change identical.

I did this because I didn't know the details of the code around here and couldn't assess the side effect of deleting the JSON auto-conversion at that moment. If it doesn't introduce errors, I think we can delete this now.

Reply:
Maybe the care we should take care of is where the return value includes dataframe or numpy array?
I think those were JSON encoded by `orjson` under the hood.

Reply:
As far as I can see, we don't return dataframes or numpys directly from any postprocess function. However I switched to using orjson.dumps() anyways, I believe this should have strictly better performance.